src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -152,52 +91,82 @@ def __init__(
         self._run_id = run_id
         self._variant_id = variant_id
         self._validate_inputs = validate_inputs
-        multiprocessing_start_method = os.environ.get("PF_BATCH_METHOD")
         sys_start_methods = multiprocessing.get_all_start_methods()
-        if multiprocessing_start_method and multiprocessing_start_method not in sys_start_methods:
-            bulk_logger.warning(
-                f"Failed to set start method to '{multiprocessing_start_method}', "
-                f"start method {multiprocessing_start_method} is not in: {sys_start_methods}."
-            )
-            bulk_logger.info(f"Set start method to default {multiprocessing.get_start_method()}.")
-            multiprocessing_start_method = None
-        self.context = get_multiprocessing_context(multiprocessing_start_method)
-        use_fork = self.context.get_start_method() == "fork"
-        # When using fork, we use this method to create the executor to avoid reloading the flow
-        # which will introduce a lot more memory.
-        if use_fork:
-            self._executor_creation_func = partial(create_executor_fork, flow_executor=flow_executor)
-        elif flow_executor._flow_file:
-            self._executor_creation_func = partial(
-                FlowExecutor.create,
-                flow_file=flow_executor._flow_file,
-                connections=flow_executor._connections,
-                working_dir=flow_executor._working_dir,
-                raise_ex=False,
-            )
-        else:  # Legacy flow executor, will be deprecated with the legacy pf portal.
-            self._executor_creation_func = partial(
-                create_executor_legacy,
-                flow=flow_executor._flow,
-                connections=flow_executor._connections,
-                loaded_tools=flow_executor._loaded_tools,
-                cache_manager=flow_executor._cache_manager,
-            )
+        use_fork = "fork" in sys_start_methods
+        self.context = get_multiprocessing_context("fork" if use_fork else "spawn")
+        self._flow_file = flow_executor._flow_file
+        self._connections = flow_executor._connections
+        self._working_dir = flow_executor._working_dir
+
+        if not use_fork:
```

Comment:
Just a stupid question. You don't have to modify.

Why we keep the self._executor_creation_func in __init__ func? Can we simply put the logic in __enter__ and remove the self._executor_creation_func?